Firestore/core/src/firebase/firestore/core/query.cc

```diff
@@ -27,6 +31,25 @@ using model::Document;
 using model::DocumentKey;
 using model::ResourcePath;
 
+namespace {
+
+// Convert a vector of unique_ptr's to a vector of shared_ptr's
+std::vector<std::shared_ptr<core::Filter>> Convert(
+    std::vector<std::unique_ptr<core::Filter>>&& v) {
+  std::vector<std::shared_ptr<core::Filter>> result;
+  std::move(v.begin(), v.end(), std::back_inserter(result));
+  v.clear();
+  return result;
+}
+
+}  // namespace
+
+Query::Query(ResourcePath path,
+             std::vector<std::unique_ptr<core::Filter>>&&
```

Comment:
If this is the only overload of the Query constructor, an rvalue reference here is fine, since it forces the caller to std::move. If you're at all tempted to add another overload that takes a `const std::vector<pointer_type<core::Filter>>&` then prefer to just pass by value here.

Reply:
With the unique/shared ptr split, I like forcing std::move. Without, I don't, since it allows me to eliminate the private ctor.